File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Linux/X64/core/receiver/receiver.rb
require_relative '../interpreter/interpreter'

class Receiver
  
  def initialize
    @@interpreter = Interpreter.new
  end

  def send_command(message_array, message_array_len)
    @@interpreter.process(message_array, message_array_len)
  end
  
  def heart_beat(message_array, message_array_len)
    [49,48]
  end
end

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/lib/core/receiver/Receiver.js
const { InMemoryConnectionData, CommandSerializer, Interpreter } = require('../../..')

let RuntimeLogger

class Receiver {
    static connectionData = new InMemoryConnectionData()
    Receiver() {
        if (!RuntimeLogger) {
            RuntimeLogger = require('../../utils/RuntimeLogger')
        }
        RuntimeLogger.printRuntimeInfo()
    }

    /**
     * @param {number[]} messageByteArray
     */
    static sendCommand(messageByteArray) {
        return new CommandSerializer().serialize(
            new Interpreter().process(messageByteArray),
            this.connectionData
        )
    }

    static heartBeat(messageByteArray) {
        let response = new Int8Array(2)
        response[0] = messageByteArray[11]
        response[1] = messageByteArray[12] - 2
        return response
    }
}

module.exports = Receiver
